src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java

```diff
@@ -144,12 +193,264 @@ public String getApiUri() {
             }
+        } catch(final Exception e) {
+            response.put("error", e.getMessage());
         }
+
+        return respondJson(500, response.toString());
     }
 
-    private void add(TaskListener listener, SCMSourceObserver observer, GHRepository repo) throws InterruptedException {
+    /**
+     * Create a github usesrname/password credential with the apiKey
+     */
+    private StandardCredentials createCredentials(String apiUrl, String username, String apiKey, String identifier) {
+        String description = String.format("%s GitHub auth token *%s", username, identifier);
+        final UsernamePasswordCredentialsImpl creds = new UsernamePasswordCredentialsImpl(
+                CredentialsScope.GLOBAL,
+                UUID.randomUUID().toString(),
+                description,
+                username,
+                apiKey);
+
+        URI serverUri = URI.create(apiUrl = StringUtils.defaultIfBlank(apiUrl, Connector.GITHUB_CREDENTIAL_DOMAIN_URI));
+
+        List<DomainSpecification> specifications = Arrays.asList(
+            new SchemeSpecification(serverUri.getScheme()),
+            new HostnameSpecification(serverUri.getHost(), null)
+        );
+
+        final Domain domain = new Domain(serverUri.getHost(), serverUri.getHost(), specifications);
+        ACL.impersonate(ACL.SYSTEM, new Runnable() { // do it with system rights
+            @Override
+            public void run() {
+                try {
+                    new SystemCredentialsProvider.StoreImpl().addDomain(domain, creds);
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        });
+
+        return creds;
+
+    }
+
+    private HttpResponse respondJson(final int status, final String json) {
+        return new HttpResponse() {
+            @Override
+            public void generateResponse(StaplerRequest req, StaplerResponse rsp, Object node) throws IOException, ServletException {
+                rsp.setStatus(status);
+                rsp.setContentType("application/json;charset=UTF-8");
+                rsp.getWriter().println(json);
+            }
+        };
+    }
+
+    /**
+     * Registers appropriate web hooks with GitHub
+     */
+    private void registerOrganizationWebhook(SCMSourceObserver observer, GHRepository repo, String apiUrl) throws Exception {
+        StandardCredentials credentials = Connector.lookupScanCredentials(observer.getContext(), apiUrl, scanCredentialsId);
+        UsernamePasswordCredentials creds = (UsernamePasswordCredentials)credentials;
+
+        String jenkinsUrl = Jenkins.getActiveInstance().getRootUrl();
+
+        if (isValidExternalJenkinsUrl(jenkinsUrl)) {
+            observer.getListener().getLogger().format("Skipping webhook registration for invalid external URL: " + jenkinsUrl + "%n");
+            return;
+        }
+
+        if (!jenkinsUrl.endsWith("/")) {
```

Comment:
AFAIK `Jenkins.getActiveInstance().getRootUrl()` returns always the trailing slash.